tcpproxy.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	"io"
+	"net"
+	"sync"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+// TCPProxy is a simple TCP proxy that can be used
```

Comment:
Not sure. I can't think of a decent name for that, and the basic structure might well be useful for other things in the future too. The fact that Close closes existing connections isn't really unexpected for a proxy, I'd've thought.